python-lib/tc_etl_lib/tc_etl_lib/cb.py

```diff
@@ -85,13 +86,16 @@ def __init__(self,*, endpoint: Optional[str] = None, timeout: int = 10, post_ret
         self.sleep_send_batch = sleep_send_batch
         self.cb_flowcontrol = cb_flowcontrol
         self.batch_size = batch_size
+        self.session = requests.Session()
 
         # check block_size limit.
         if (int(block_size) > int(800000)):
             raise ValueError('Block size limit reached! <<block_size>> value cannot be greater than 800000')
 
         self.block_size = block_size
 
+    def __del__(self):
+        self.session.close()
```

Comment:
Tal vez puesto a controlar las excepciones de cierre, he leído por algún lado que el `__del__`  no es inmediato y que se puede encontrar que el recurso a cerrar ya ande destruido (saltando una excepción), he visto implementaciones que incorporan un try en el cierre, para controlar esos casos puntuales, o incluso un control por None con un `if self.session is not None` para controlar que no esté ya cerrado en ese momento.

```suggestion
        try:
            self.session.close()
        except Exception:  # pylint: disable=broad-except
            pass
```